### PR TITLE
Bump MINIMAP2_INDEX container versions

### LINKED ISSUE #2444: MINIMAP2_ALIGN and _INDEX have different versions of minimap in their default containers
### Have you checked the docs?

- [X] [nf-core website: troubleshooting](https://nf-co.re/usage/troubleshooting)
- [X] [nf-core modules documentation](https://nf-co.re/docs/contributing/modules)

### Description of the bug

The `MINIMAP2_INDEX` and `MINIMAP2_ALIGN` containers are different, as `_ALIGN` uses `samtools` for `bam` generation and needs a `mulled` container, whilst `_INDEX` uses only `minimap2`. 

This has resulted in the two modules having different versions of `minimap2` by default. This does not seem ideal if the intention is to pass the output of `_INDEX` into `_ALIGN`.

```
MINIMAP2_ALIGN:
  minimap2: 2.24-r1122
MINIMAP2_INDEX:
  minimap2: 2.21-r1071
```

I would propose either using the same `mulled` container for `_INDEX` despite not requiring `samtools`?

```console
│ minimap2/align              │ https://github.com/nf-core/modules.git │ 5850432aab24a1924389b660adfee3809d3e60a9 │ Allow self-alignment of reads in minimap2 (#2342) │ 2022-10-21 │
│ minimap2/index              │ https://github.com/nf-core/modules.git │ 5e34754d42cd2d5d248ca8673c0a53cdf5624905 │ Restructure nf-core/modules repo (#2141)          │ 2022-10-04 │
```

### Relevant files

_No response_

### System information

_No response_

### PR DESCRIPTION
**Pull request by awgymer**
MINIMAP2_ALIGN has a newer version of minimap2 in its container than MINIMAP2_INDEX. 
They should be aligned.

## PR checklist

Closes #2444  

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/minimap2/index/main.nf [hunk 1]
```diff
@@ -3,8 +3,8 @@ process MINIMAP2_INDEX {
 
     conda (params.enable_conda ? 'bioconda::minimap2=2.21' : null)
```
Comment (muffato) [inline modules/nf-core/minimap2/index/main.nf:4]:
```suggestion
    // Keep the same as in minimap2/align
    conda (params.enable_conda ? 'bioconda::minimap2=2.21' : null)
```

Comment (muffato) [inline modules/nf-core/minimap2/index/main.nf:4]:
```suggestion
    // Note: the versions here need to match the versions used in minimap2/align
    conda (params.enable_conda ? 'bioconda::minimap2=2.24' : null)
```

Comment (muffato) [reply modules/nf-core/minimap2/index/main.nf:4]: And https://github.com/nf-core/modules/blob/42a5fb17f30c047edd307a73f900b0fabb7df3ae/modules/nf-core/minimap2/align/main.nf#L5 should be updated to match the versions of minimap2 and samtools used in the mulled container + a note to keep the versions in sync with minimap2/index

**Review by muffato — CHANGES_REQUESTED** [review]

File: modules/nf-core/minimap2/align/main.nf [hunk 2]
```diff
@@ -2,7 +2,7 @@ process MINIMAP2_ALIGN {
     tag "$meta.id"
     label 'process_medium'
 
-    conda (params.enable_conda ? 'bioconda::minimap2=2.21 bioconda::samtools=1.12' : null)
+    conda (params.enable_conda ? 'bioconda::minimap2=2.24 bioconda::samtools=1.14' : null)
```
Comment (muffato) [inline modules/nf-core/minimap2/align/main.nf:5]:
```suggestion
    // Note: the versions here need to match the versions used in the mulled container below and minimap2/index
    conda (params.enable_conda ? 'bioconda::minimap2=2.24 bioconda::samtools=1.14' : null)
```

**Comment by awgymer** [comment]
Header from docker test (on my mac):
```
@HD	VN:1.6	SO:coordinate
@SQ	SN:MT192765.1	LN:29829
@PG	ID:minimap2	PN:minimap2	VN:2.24-r1122	CL:minimap2 -t 2 -a genome.fasta test_1.fastq.gz test_2.fastq.gz
@PG	ID:samtools	PN:samtools	PP:minimap2	VN:1.14	CL:samtools sort
@PG	ID:samtools.1	PN:samtools	PP:samtools	VN:1.14	CL:samtools view -@ 2 -b -h -o test.bam
```

Header from conda test (on GitHub):
```
@HD	VN:1.6	SO:coordinate
@SQ	SN:MT192765.1	LN:29829
@PG	ID:minimap2	PN:minimap2	VN:2.24-r1122	CL:minimap2 -t 2 -a genome.fasta test_1.fastq.gz test_2.fastq.gz
@PG	ID:samtools	PN:samtools	PP:minimap2	VN:1.14	CL:samtools sort
@PG	ID:samtools.1	PN:samtools	PP:samtools	VN:1.14	CL:samtools view -@ 2 -b -h -o test.bam
```

Failing test doesn’t seem to be the headers which is normally where I find the issue

**Comment by muffato** [comment]
I've seen that in many other modules. The uncompressed BAM files are identical, but the compressed versions differ.

**Pull request merged**